mailpoet/lib/Newsletter/NewsletterValidator.php

```diff
@@ -0,0 +1,104 @@
+<?php declare(strict_types=1);
+
+namespace MailPoet\Newsletter;
+
+use MailPoet\Entities\NewsletterEntity;
+use MailPoet\Services\Bridge;
+use MailPoet\Settings\TrackingConfig;
+use MailPoet\Util\License\Features\Subscribers as SubscribersFeature;
+use MailPoet\Validator\ValidationException;
+
+class NewsletterValidator {
+  
+  /** @var Bridge */
+  private $bridge;
+
+  /** @var TrackingConfig */
+  private $trackingConfig;
+
+  /** @var SubscribersFeature */
+  private $subscribersFeature;
+
+  public function __construct(
+    Bridge $bridge,
+    TrackingConfig $trackingConfig,
+    SubscribersFeature $subscribersFeature
+  ) {
+    $this->bridge = $bridge;
+    $this->trackingConfig = $trackingConfig;
+    $this->subscribersFeature = $subscribersFeature;
+  }
+  
+  public function validate(NewsletterEntity $newsletterEntity): ?string {
+    try {
+      $this->validateSubscriberLimit();
```

Comment:
I don't know if we need to address this in this PR, but maybe a solution to the problem that you are raising, that would improve the user experience in my opinion, is to return all the errors that we find instead of just the first error.

Reply:
That occurred to me as well, though sometimes that approach can be extra noisy because one error might always imply another (e.g. if the email doesn't have any content, you might also see an error about it missing an unsubscribe link, which is kind of redundant). I'll leave this as-is for now because it doesn't seem like a major issue.

Reply:
:+1: